echopype/mask/single_target_detection/detect_matecho.py

```diff
@@ -0,0 +1,1005 @@
+        if n == 0:
+            return np.asarray([], dtype=dtype)
+        a = np.asarray(vals)
+        if dtype is not None:
+            a = a.astype(dtype)
+        if a.size != n:
+            raise ValueError(f"Output field '{key}' has length {a.size}, expected {n}.")
+        return a
+
+    tsfreq = np.asarray(out.get("TSfreq_matrix", np.empty((0, 23))), dtype=float)
+    if tsfreq.ndim == 1:
+        tsfreq = tsfreq.reshape(1, -1)
+    if tsfreq.size == 0:
+        tsfreq = np.empty((0, 23), dtype=float)
+    if tsfreq.shape[0] != n:
+        raise ValueError(
+            f"TSfreq_matrix has {tsfreq.shape[0]} rows but expected {n} (nb_valid_targets)."
+        )
+
+    ds_out = xr.Dataset(
+        data_vars=dict(
+            nb_valid_targets=((), n),
+            TS_comp=("target", _arr("TS_comp", float), {"units": "dB"}),
+            TS_uncomp=("target", _arr("TS_uncomp", float), {"units": "dB"}),
+            Target_range=("target", _arr("Target_range", float), {"units": "m"}),
+            Target_range_disp=("target", _arr("Target_range_disp", float), {"units": "m"}),
+            Target_range_min=("target", _arr("Target_range_min", float), {"units": "m"}),
+            Target_range_max=("target", _arr("Target_range_max", float), {"units": "m"}),
+            idx_r=("target", _arr("idx_r", int)),
+            idx_target_lin=("target", _arr("idx_target_lin", int)),
+            pulse_env_before_lin=("target", _arr("pulse_env_before_lin", float)),
+            pulse_env_after_lin=("target", _arr("pulse_env_after_lin", float)),
+            PulseLength_Normalized_PLDL=("target", _arr("PulseLength_Normalized_PLDL", float)),
+            Transmitted_pulse_length=("target", _arr("Transmitted_pulse_length", float)),
+            Angle_minor_axis=("target", _arr("Angle_minor_axis", float), {"units": "rad"}),
+            Angle_major_axis=("target", _arr("Angle_major_axis", float), {"units": "rad"}),
+            StandDev_Angles_Minor_Axis=(
+                "target",
+                _arr("StandDev_Angles_Minor_Axis", float),
+                {"units": "phase_steps"},
+            ),
+            StandDev_Angles_Major_Axis=(
+                "target",
+                _arr("StandDev_Angles_Major_Axis", float),
+                {"units": "phase_steps"},
+            ),
+            Heave=("target", _arr("Heave", float), {"units": "m"}),
+            Roll=("target", _arr("Roll", float), {"units": "rad"}),
+            Pitch=("target", _arr("Pitch", float), {"units": "rad"}),
+            Heading=("target", _arr("Heading", float), {"units": "rad"}),
+            Dist=("target", _arr("Dist", float), {"units": "m"}),
+            TSfreq_matrix=(("target", "tsfreq_col"), tsfreq, {}),
+        ),
+        coords=dict(
+            target=target,
+            tsfreq_col=np.arange(tsfreq.shape[1], dtype=int),
+            Ping_number=("target", _arr("Ping_number", int)),
+            Time=("target", _arr("Time", np.dtype("datetime64[ns]"))),
```

Comment:
I don't see any variables above aligned with `Ping_number` or `Time`, so think these are redundant and can be removed?

Also, ping number and time are themselves the same thing, unless the timestamps are not derived from ping time (though I think for ping-by-ping methods like this it _is_ derived from ping time?)